src/Components/Components/src/PublicAPI.Unshipped.txt

```diff
@@ -29,6 +31,15 @@ Microsoft.AspNetCore.Components.Lifetime.ComponentApplicationLifetime.State.get
 Microsoft.AspNetCore.Components.Lifetime.IComponentApplicationStateStore
 Microsoft.AspNetCore.Components.Lifetime.IComponentApplicationStateStore.GetPersistedStateAsync() -> System.Threading.Tasks.Task<System.Collections.Generic.IDictionary<string!, byte[]!>!>!
 Microsoft.AspNetCore.Components.Lifetime.IComponentApplicationStateStore.PersistStateAsync(System.Collections.Generic.IReadOnlyDictionary<string!, byte[]!>! state) -> System.Threading.Tasks.Task!
+Microsoft.AspNetCore.Components.NavigationManager.NavigateTo(string! uri, Microsoft.AspNetCore.Components.NavigationOptions options) -> void
+Microsoft.AspNetCore.Components.NavigationManager.NavigateTo(string! uri, bool forceLoad = false, bool replace = false) -> void
```

Comment:
Do we need this overload?

I think we should force people into using the navigation options directly and set ourselves for success in the future.

Reply:
Personally I find:

```cs
navManager.NavigateTo("something", replace: true);
```

to be a lot more readable than:

```cs
navManager.NavigateTo("something", new NavigationOptions { ReplaceHistoryEntry = true });
```

However I understand it's always a subjective balance when we consider adding overloads just for convenience. It depends on how universal we think the convenience-overload pattern will be, and whether it will stand the test of time as being the only extra convenience we need, vs something that ends up getting multiplied out into many other overload variants. (I know you know this; just giving context for anyone else reading)

We do need at least to keep the `NavigateTo(string, bool)` overload for back-compat. To me it felt a bit rough if this other equally-often-used flag was *not* available in a similar shorthand format.

Shall we leave a final call on this until we get to API review, or do you feel strongly now?

Reply:
I don't feel strongly about it, it was mainly a suggestion.